libraries/AP_Mount/AP_Mount.cpp

```diff
@@ -178,6 +179,15 @@ void AP_Mount::init()
             serial_instance++;
             break;
 #endif // HAL_MOUNT_CADDX_ENABLED
+
+#if HAL_MOUNT_XFROBOT_ENABLED
+        // check for XFRobot gimbal
+        case Type::XFRobot:
+            _backends[instance] = NEW_NOTHROW AP_Mount_XFRobot(*this, _params[instance], instance, serial_instance);
```

Comment:
i think the num_instances shouldn't increment on alloc failure, pattern is to do it outside of the switch